lib/system/alloc.nim

```diff
@@ -943,12 +943,13 @@ proc rawDealloc(a: var MemRegion, p: pointer) =
                 s -% sizeof(FreeCell))
       # check if it is not in the freeSmallChunks[s] list:
       if c.free < s:
+        # TODO: Steal freeList from the other chunk? Leaving it as is will waste a lot of cells until we return to the other chunk
         # add it to the freeSmallChunks[s] array:
         listAdd(a.freeSmallChunks[s div MemAlign], c)
         inc(c.free, s)
       else:
         inc(c.free, s)
-        if c.free == SmallChunkSize-smallChunkOverhead():
+        if a.freeSmallChunks[s div MemAlign] == c and c.free == SmallChunkSize-smallChunkOverhead():
```

Comment:
Change the order of these tests for better performance.

Reply:
You're right, I actually tried that locally and it didn't make a significant difference in my tests, but no reason to leave it on the table.